File: VB/ColumnSpacing/MyGridViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System.Drawing
Imports DevExpress.XtraGrid.Columns

Namespace ColumnSpacing
	Public Class MyGridViewInfo
		Inherits GridViewInfo
		Public Sub New(ByVal gridView As DevExpress.XtraGrid.Views.Grid.GridView)
			MyBase.New(gridView)
		End Sub

		Protected Overrides Sub CalcColumnsDrawInfo()
			MyBase.CalcColumnsDrawInfo()

			Dim count As Integer = View.VisibleColumns.Count + (If(AllowBehindColumn AndAlso (View.VisibleColumns.Count = 0 OrElse True), 1, 0))
			Dim firstIndex As Integer = If(ViewRects.IndicatorWidth > 0, -1, 0)
			For i As Integer = firstIndex To count-2
				Dim column As GridColumn = If(i > -1, View.GetVisibleColumn(i), Nothing)
				Dim space As Integer = 0
				If column IsNot Nothing Then
					space = (TryCast(column, MyGridColumn)).ColumnSpacing
				End If

				If (ColumnsInfo.Count <> 0) AndAlso (i >= 0) AndAlso (ColumnsInfo(i+1) IsNot Nothing) Then
					Dim rect As Rectangle = ColumnsInfo(i+1).Bounds
					If i > 0 Then
                        ColumnsInfo(i + 1).Bounds = New Rectangle(rect.Left + space, rect.Top, rect.Width - (space * 2), rect.Height)
					Else
                        ColumnsInfo(i + 1).Bounds = New Rectangle(rect.Left, rect.Top, rect.Width - space, rect.Height)
					End If
				End If
			Next i
		End Sub
	End Class
End Namespace

File: CS/ColumnSpacing/MyGridViewInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System.Drawing;
using DevExpress.XtraGrid.Columns;

namespace ColumnSpacing
{
    public class MyGridViewInfo : GridViewInfo
    {
        public MyGridViewInfo(DevExpress.XtraGrid.Views.Grid.GridView gridView) : base(gridView) { }

        protected override void CalcColumnsDrawInfo()
        {
            base.CalcColumnsDrawInfo();

            int count = View.VisibleColumns.Count + (AllowBehindColumn && (View.VisibleColumns.Count == 0 || true) ? 1 : 0);
            int firstIndex = ViewRects.IndicatorWidth > 0 ? -1 : 0;
            for (int i = firstIndex; i < count-1; i++)
            {
                GridColumn column = i > -1 ? View.GetVisibleColumn(i) : null;
                int space = 0;
                if (column != null) space = (column as MyGridColumn).ColumnSpacing;

                if ((ColumnsInfo.Count != 0) && (i >= 0) && (ColumnsInfo[i+1] != null))
                {
                    Rectangle rect = ColumnsInfo[i+1].Bounds;
                    if (i > 0)
                        ColumnsInfo[i + 1].Bounds = new Rectangle(rect.Left + space, rect.Top, rect.Width - (space * 2), rect.Height);
                    else
                        ColumnsInfo[i + 1].Bounds = new Rectangle(rect.Left, rect.Top, rect.Width - (space)-1, rect.Height);
                }
            }
        }
    }
}
